List recurring events on android.

snovak opened the pull request:
No changes to the API.  This change also includes recurring events when requesting events from the API, on android.  This partially addresses functionality requested in https://github.com/ebarooni/capacitor-calendar/issues/99
Other changes to package versions/options were necessary to get the library to build.

ebarooni commented:
@snovak This is weird I have not seen this before. What node version do you use? Can you please delete your `node_modules` and undo your `package-lock.json` file and install the dependencies with `npm ci`?

There is currently another PR and they also did not report such issue. I think your node version is causing issues here. Im using `20.12.2`

snovak commented:
I was using latest stable v23.3.0.  I changed to 20.12.2 with nvm.  Same issue.  

```
snovak@dad:~/Projects/nag.bot/app/plugins/capacitor-calendar$ npm ci

added 326 packages, and audited 327 packages in 3s

126 packages are looking for funding
  run `npm fund` for details

6 low severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
snovak@dad:~/Projects/nag.bot/app/plugins/capacitor-calendar$ npm list
@ebarooni/capacitor-calendar@7.2.0 /home/snovak/Projects/nag.bot/app/plugins/capacitor-calendar
├── @capacitor/android@7.0.1
├── @capacitor/core@7.0.1
├── @capacitor/ios@7.0.1
├── @ebarooni/capacitor-docgen@0.0.2
├── @ebarooni/eslint-config@1.3.0
├── @ebarooni/prettier-config@1.4.0
├── @ebarooni/swiftlint-config@1.2.1
├── @eslint/js@9.19.0
├── @eslint/json@0.10.0
├── @eslint/markdown@6.2.2
├── angular-eslint@19.0.2
├── eslint-config-prettier@10.0.1
├── eslint-plugin-promise@7.2.1
├── eslint@9.19.0
├── prettier@3.4.2
├── rimraf@6.0.1
├── rollup@4.32.1
├── swiftlint@2.0.0
├── typedoc@0.26.5
├── typescript-eslint@8.22.0
└── typescript@4.8.4

snovak@dad:~/Projects/nag.bot/app/plugins/capacitor-calendar$ node -v
v20.12.2
snovak@dad:~/Projects/nag.bot/app/plugins/capacitor-calendar$ npm run build

> @ebarooni/capacitor-calendar@7.2.0 build
> npm run clean && npm run docgen && tsc && rollup -c rollup.config.mjs


> @ebarooni/capacitor-calendar@7.2.0 clean
> rimraf ./dist


> @ebarooni/capacitor-calendar@7.2.0 docgen
> docgen --api CapacitorCalendarPlugin --output-readme README.md --output-json dist/docs.json


✔️ DocGen Output: /home/snovak/Projects/nag.bot/app/plugins/capacitor-calendar/dist/docs.json
✔️ DocGen Output: /home/snovak/Projects/nag.bot/app/plugins/capacitor-calendar/README.md

../../node_modules/@types/d3-dispatch/index.d.ts:91:5 - error TS1139: Type parameter declaration expected.

91     const EventNames extends keyof any = keyof EventMap,
       ~~~~~

../../node_modules/@types/d3-dispatch/index.d.ts:91:22 - error TS1005: ',' expected.

91     const EventNames extends keyof any = keyof EventMap,
                        ~~~~~~~

../../node_modules/@types/d3-dispatch/index.d.ts:91:36 - error TS1005: ',' expected.

91     const EventNames extends keyof any = keyof EventMap,
                                      ~~~

../../node_modules/@types/d3-dispatch/index.d.ts:91:48 - error TS1005: ',' expected.

91     const EventNames extends keyof any = keyof EventMap,
                                                  ~~~~~~~~

../../node_modules/@types/d3-dispatch/index.d.ts:92:1 - error TS1109: Expression expected.

92 >(
   ~

../../node_modules/@types/d3-dispatch/index.d.ts:93:5 - error TS1109: Expression expected.

93     ...types: EventNames[]
       ~~~

../../node_modules/@types/d3-dispatch/index.d.ts:93:26 - error TS1011: An element access expression should take an argument.

93     ...types: EventNames[]
                            

../../node_modules/@types/d3-dispatch/index.d.ts:94:1 - error TS1128: Declaration or statement expected.

94 ): Dispatch<This, EventMap>;
   ~

../../node_modules/@types/d3-dispatch/index.d.ts:94:2 - error TS1128: Declaration or statement expected.

94 ): Dispatch<This, EventMap>;
    ~


Found 9 errors in the same file, starting at: ../../node_modules/@types/d3-dispatch/index.d.ts:91
```

ebarooni commented:
@snovak Your build errors seem to be unrelated to this project. `../../node_modules/@types/d3-dispatch/index.d.ts:91:5` is two layers above the directory of the plugin. Are you using d3 by any chance in your project?

d3 is not a dependency of this plugin so I doubt the error is related to it.
